gift-certificate-application/controller/src/main/java/com/epam/esm/controller/OrderController.java

```diff
@@ -48,15 +51,16 @@ public Order create(@RequestBody OrderDto orderDto) {
     }
 
     @DeleteMapping("/{id}")
+    @PreAuthorize("hasAuthority('orders:delete')")
     @ResponseStatus(HttpStatus.NO_CONTENT)
     public void delete(@PathVariable Long id) {
         orderService.delete(id);
     }
 
-    @GetMapping
-    public List<Order> findByAttributes(@RequestParam(required = false, name = "user-id") Long userId,
-                                    CustomPagination pagination) {
-        List<Order> orderList = orderService.findByAttributes(userId, pagination);
+    @GetMapping("/user-id/{userId}")
```

Comment:
This URL violates the principles of the REST API, the user ID is not a resource.
This URL might look like /users/{usedId}/orders (preferred) or /orders?userId=